### ISSUE TITLE
power density calculation based on ray tracing results

### ISSUE THREAD

**Issue by yangfg-bsrf** [kklmn/xrt#76]
Hi, now I want to evaluate the heating-load on the mirror.  The beam data for the following image has been exported , then I am not sure how to calculate the power density profile, because the N_good is  equal to the N_all. Can you give me some advices? Thanks!

![B3_HKBLocal](https://user-images.githubusercontent.com/41823786/114993326-48659300-9ece-11eb-94f1-98f329a1f6bd.png)

**Comment by kklmn** [comment]
Hi,
Please see [this](https://xrt.readthedocs.io/gallery2.html#diamond-filter-of-varying-thickness) example for how to make power plots with power density isolines. Search the examples for `fluxKind='power'` in py files.

**Comment by yangfg-bsrf** [comment]
yea, the total power is shown in the result. 
![B3_HKBLocal -0 00196](https://user-images.githubusercontent.com/41823786/115035545-42d47100-9eff-11eb-96e1-80edd756d4d9.png)

Further, I want to generate the map of the power density distribution along the mirror.  I am confused by the codes in  XYCPlot. can you tell me which part  code  is for the  calculation for self.intensity.
> ` def _get_flux(self):
>         self.flux = float(self.intensity) / self.nRaysAll *\
>             self.nRaysSeededI / self.nRaysSeeded`
> 
>     `def _get_power(self):
>         self.power = self.intensity / self.nRaysAll`

**Comment by kklmn** [comment]
> Further, I want to generate the map of the power density distribution along the mirror. 

The 2D and 1D histograms give you this distribution as plot.total2D and ax.total1D for ax in [plot.xaxis, plot.yaxis, plot.caxis]. The normalization constant = plot.power. Do you need something more?

**Comment by yangfg-bsrf** [comment]
we want to evaluate the thermal deformation of  the mirror, so the 2D heating-load distribution is necessary. For this case, I believe  plot.total2D is enough because the energy bandwidth is small. does it consider the power difference among the rays with different energy? that is why I want to know how the intensity is calculated.

**Comment by kklmn** [comment]
> does it consider the power difference among the rays with different energy?

No. This would need 3D histogramming, which we don't do. If you need it, you should compose it manually from 2D histograms.

>that is why I want to know how the intensity is calculated.

`plot.intensity` is only the sum of ray intensity values (Jss^2 + Jpp^2) over the 2D histogram. The name came from Shadow. The physical intensity (the flux density) is calculated by 2D histogramming in the module xrt.multipro.py
